src/lib/configs.ts

```diff
@@ -38,16 +44,30 @@ async function loadConfigsFromFile() {
     if (typeof projectConfigs === 'function') projectConfigs = await projectConfigs()
     if (!projectConfigs) return
 
-    GLEE_DIR = projectConfigs.GLEE_DIR || GLEE_DIR 
-    GLEE_LIFECYCLE_DIR = projectConfigs.GLEE_LIFECYCLE_DIR ? path.resolve(GLEE_DIR, projectConfigs.GLEE_LIFECYCLE_DIR) : GLEE_LIFECYCLE_DIR 
-    GLEE_FUNCTIONS_DIR = projectConfigs.GLEE_FUNCTIONS_DIR ? path.resolve(GLEE_DIR, projectConfigs.GLEE_FUNCTIONS_DIR) : GLEE_FUNCTIONS_DIR 
-    ASYNCAPI_FILE_PATH = projectConfigs.ASYNCAPI_FILE_PATH ? path.resolve(GLEE_DIR, projectConfigs.ASYNCAPI_FILE_PATH) : ASYNCAPI_FILE_PATH
+    GLEE_DIR = projectConfigs.glee?.gleeDir || GLEE_DIR 
+    GLEE_LIFECYCLE_DIR = projectConfigs.glee?.lifecycleDir ?? GLEE_LIFECYCLE_DIR 
+    GLEE_FUNCTIONS_DIR = projectConfigs.glee?.functionsDir ?? GLEE_FUNCTIONS_DIR 
+    ASYNCAPI_FILE_PATH = projectConfigs.glee?.asyncapiFilePath ?? ASYNCAPI_FILE_PATH
     return projectConfigs
   } catch (e) {
     return console.error(e)
   }
 }
 
+export function findSpecFile(baseDir: string): string{
+  const files = ['asyncapi.yaml', 'asyncapi.json', 'asyncapi.yml']
+  const foundFiles = files.filter(file => existsSync(path.resolve(baseDir, file)))
```

Comment:
You should not be using `existsSync` because:

1. It only tests the existence of the path but it doesn't test if you have read permission. So it may happen that the file is there but you can't open it.
2. It tests the existence of the **path**. A path might be a file or a directory. We want to make sure it's a file.

Therefore I recommend using [fs.accessSync](https://nodejs.org/api/fs.html#fsaccesssyncpath-mode) and [fs.statSync](https://nodejs.org/api/fs.html#fsstatsyncpath-options) (which allows you to check for [isFile()](https://nodejs.org/api/fs.html#statsisfile)).